libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -693,20 +693,6 @@ IotNetworkError_t IotNetworkAfr_Destroy( void * pConnection )
     }
     else
     {
-        /* As this function should be called ONLY called after the connection is closed,
-         * the receive task should have already exited. */
-        if( pNetworkConnection->receiveCallback != NULL )
```

Comment:
It will be valuable to retain this check for ensuring that the receive task has been deleted before the socket resources are cleared.
Instead of removing this assert check, can you update the `IotNetworkAfr_Close` function to _not_ clear the `_FLAG_RECEIVE_TASK_EXITED` flag?

Reply:
Updated